src/lib.rs

```diff
@@ -126,21 +126,21 @@ pub fn murmur3_x86_128<T :Read>(source: &mut T, seed: u32, out: &mut [u8]) {
                     9...11 => {
                         h1 ^= process_h1_k_x86(LittleEndian::read_u32(&buf[0..4]));
                         h2 ^= process_h2_k_x86(LittleEndian::read_u32(&buf[4..8]));
-                        h3 ^= process_h3_k_x86(LittleEndian::read_int(&buf[8..size], size -8) as u32);
+                        h3 ^= process_h3_k_x86(LittleEndian::read_uint(&buf[8..size], size -8) as u32);
                     }
                     8 => {
                         h1 ^= process_h1_k_x86(LittleEndian::read_u32(&buf[0..4]));
                         h2 ^= process_h2_k_x86(LittleEndian::read_u32(&buf[4..8]));
                     }
                     5...7 => {
                         h1 ^= process_h1_k_x86(LittleEndian::read_u32(&buf[0..4]));
-                        h2 ^= process_h2_k_x86(LittleEndian::read_int(&buf[4..size], size-4) as u32);
+                        h2 ^= process_h2_k_x86(LittleEndian::read_uint(&buf[4..size], size-4) as u32);
                     }
                     4 => {
                         h1 ^= process_h1_k_x86(LittleEndian::read_u32(&buf));
                     }
                     1...3 => {
-                        h1 ^= process_h1_k_x86(LittleEndian::read_int(&buf, size) as u32);
+                        h1 ^= process_h1_k_x86(LittleEndian::read_uint(&buf, size) as u32);
```

Comment:
So why `uint` instead of `u32`?

Reply:
`read_u32` will read 4 bytes of `buf`, but the `buf` may contain dirty number from the previous round.